pkg/arvo/sys/vane/ames.hoon

```diff
@@ -1434,6 +1516,57 @@
     ^+  event-core
     %-  %^  trace  odd.veb  sndr.packet
         |.("received packet")
+    :: add the ship to our sponsees map if we are the sponsor
+    ::
+    =*  relay-state  relay-state.ames-state
+    =?  sponsees.relay-state
+      ?&
+        =(~ next.packet)
+        =/  val  (~(get by sponsees.relay-state) sndr.packet)
+        =(~ val)
+        =(our (sein:title our now sndr.packet))
+      ==
+      ::  send 30s expiration timer to behn
+      ::
+      =/  duct  ~[/ames]
+      =/  wire  (make-sponsee-timer-wire sndr.packet)
+      =.  event-core
+        (emit duct %pass wire %b %wait (add ~s30 now))
+      ::
+      ^+  sponsees.relay-state
+      (~(put by sponsees.relay-state) [sndr.packet [lane (add ~s30 now)]])
+    :: add the lane to our pending map
+    ::
+    =*  inv-req  [%ames `dyad`[rcvr.packet sndr.packet]]
+    =?  pending.relay-state
+      =/  val  (~(get by pending.relay-state) inv-req)
+      ?~  val  %.y
+        =/  next  q.-.u.val
+        ?~  next
+          %.n
+        =(lane u.next)
+      ::  send 30s expiration timer to behn
+      ::
+      =/  duct  ~[/ames]
+      =/  wire  (make-pending-timer-wire [rcvr.packet sndr.packet])
+      =.  event-core
+        (emit duct %pass wire %b %wait (add ~s30 now))
+      ::
+      ^+  pending.relay-state
+      (~(put by pending.relay-state) [inv-req [[lane ~] (add ~s30 now)]])
+    ::  update the pending map 'next' lane if necessary
+    ::
+    =?  pending.relay-state
+      ::
+      =/  val  (~(get by pending.relay-state) inv-req)
+      ?~  val  !!
+      =/  next  q.-.u.val
+      &(=(~ next) !=(~ next.packet))
```

Comment:
this line looks like it'll cause an early return from the `=?`, so the code on the next few lines is orphaned

Reply:
I think this is fine; this block returns the loobean for the conditional.

Reply:
ah, I see -- it should be indented, then, along with the first arg to the `=?`, like this:
```hoon
=?    pending.relay-state
    ::
    =/  val ...
    &(...)
  ::
  ^+  pending.relay-state
  ...
```